flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -182,24 +180,14 @@ public StreamGraphGenerator setJobName(String jobName) {
 		return this;
 	}
 
-	public StreamGraphGenerator setGlobalDataExchangeMode(GlobalDataExchangeMode globalDataExchangeMode) {
-		this.globalDataExchangeMode = globalDataExchangeMode;
-		return this;
-	}
-
 	public void setSavepointRestoreSettings(SavepointRestoreSettings savepointRestoreSettings) {
 		this.savepointRestoreSettings = savepointRestoreSettings;
 	}
 
 	public StreamGraph generate() {
+		runtimeExecutionMode = determineExecutionMode(runtimeExecutionMode);
```

Comment:
I think we should have a separate field and not modify the original one because that can make it hard to debug things. We could even just have a local `boolean` field that says `isBounded` because those are the only two options we have. There can be no `AUTOMATIC` from this point on. We we then also not need to have the various `verifyRuntimeModeIsSet()` checks later. WDYT?